clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -2957,7 +2957,27 @@ bool TokenAnnotator::spaceRequiredBetween(const AnnotatedLine &Line,
             FormatStyle::SBPO_ControlStatementsExceptForEachMacros &&
         Left.is(TT_ForEachMacro))
       return false;
-    return Line.Type == LT_ObjCDecl || Left.is(tok::semi) ||
+
+    if(Style.BreakConstructorInitializers == FormatStyle::BCIS_Haiku){
```

Comment:
Hi!

I think BreakConstructorInitializers is not the right variable to use here. Its documentation (https://clang.llvm.org/docs/ClangFormatStyleOptions.html) shows it is used for a different thing. It would be a bit confusing to have it affect other unrelated things.

Rather, a new "SpaceBeforeParens" style should be added (maybe named FormatStyle::SBPO_Haiku) and used here.

By cleanly separating all our changes under different options, it will be easier for other users of clang-format or haiku-format to reuse only some parts of our style, so it results in more flexibility for everyone.

Reply:
Hello,
Thank you for your review. I have made the changes according to the SpaceBeforeParens style.
Please review. 
Thanks

Reply:
Much better now, thanks!